.github/workflows/main-branch-build.yml

```diff
@@ -0,0 +1,33 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+
+name: Build for production
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:

```

Comment:
We wont be opening a pull request on main. We'll be opening a PR on a new branch, and the push to main is handled by the first condition.

Reply:
The workflow you have at the bottom will handle all other branches

Reply:
> The workflow you have at the bottom will handle all other branches

Are you referring to the test workflow here or something else?